src/script/lua_api/l_env.cpp

```diff
@@ -1381,21 +1381,13 @@ int ModApiEnv::l_override_abm(lua_State *L)
 {
 	GET_ENV_PTR;
 
-	std::string abm_name = readParam<std::string>(L, 1);
+	int abm_id = lua_tonumber(L, 1);
 
-	ActiveBlockModifier *abm = env->getActiveBlockModifierNoEx(abm_name);
-	if (!abm) {
-		throw LuaError("ABM not found.");
-		return 0;
+	LuaABM *abm = ScriptApiEnv::readABM(L, 2, abm_id);
+	if (abm != nullptr) {
+		env->setActiveBlockModifier(reinterpret_cast<ActiveBlockModifier *>(abm));
```

Comment:
i think a reinterpret cast is inappropriate here. since this is a base class C++ should just convert automatically? same as in the call to `addActiveBlockModifier`.

Reply:
It means moving the `LuaABM` definition from `s_env.cpp` back to `l_env.h`.
Done in 7ae51382c8a31a6d263ebb913c90025eaf90780c .